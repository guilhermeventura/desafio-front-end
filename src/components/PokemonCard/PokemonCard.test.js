import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { PokemonCard } from "./index";
const mockData = {
  id: 22,
  name: "pokemontest",
  sprites: { front_default: "testimg.jpg" },
  addToPokedex: () => {},
  removePokemon: () => {}
};

test("renders correctly", () => {
  const component = renderer
    .create(
      <MemoryRouter>
        <PokemonCard
          data={mockData}
          addToPokedex={mockData.addToPokedex}
          removePokemon={mockData.removePokemon}
        />
      </MemoryRouter>
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});
