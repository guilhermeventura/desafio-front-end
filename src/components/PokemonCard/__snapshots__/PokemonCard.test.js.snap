// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mdc-card pokemon__card"
>
  <a
    href="/pokemon-details/undefined"
    onClick={[Function]}
  >
    <div
      className="mdc-card__primary-action"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <h5
        className="mdc-typography mdc-typography--headline5 jovem"
      >
        pokemontest
      </h5>
      <div
        className="mdc-card__media mdc-card__media--square"
        style={
          Object {
            "backgroundImage": "url(testimg.jpg)",
          }
        }
      />
    </div>
  </a>
  <div
    className="mdc-card__actions"
  >
    <div
      className="mdc-card__action-buttons"
    >
      <button
        className="mdc-button mdc-card__action mdc-card__action--button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <span
          className="mdc-button__label"
        >
          Catch!
        </span>
      </button>
    </div>
    <div
      className="mdc-card__action-icons"
    >
      <i
        className="mdc-card__action mdc-card__action--icon"
        onClick={[Function]}
      >
        Remove
      </i>
    </div>
  </div>
</div>
`;
